### ISSUE TITLE
Starting with version 3.6.0 exception causes are not logged

### ISSUE THREAD

**Issue by sencko** [SAP/cf-java-logging-support#145]
With the latest version of **com.sap.hcp.cf.logging:cf-java-logging-support-logback** the exception cause is not logged anymore, which makes troubleshooting errors much harder. Here is some sample code to reproduce the problem:

```java

import com.sap.hcp.cf.logback.encoder.JsonEncoder;

import ch.qos.logback.classic.Level;
import ch.qos.logback.classic.Logger;
import ch.qos.logback.classic.LoggerContext;
import ch.qos.logback.classic.spi.LoggingEvent;

...
    LoggerContext context = new LoggerContext();
    Logger logger = context.getLogger(this.getClass());

    Exception cause = new Exception();
    LoggingEvent loggingEvent = new LoggingEvent(ch.qos.logback.classic.Logger.class.getName(), logger, Level.ERROR, "This is message {}",
        new Exception(cause), new Object[] { "hh" });

    JsonEncoder encoder = new JsonEncoder();
    encoder.setContext(context);
    encoder.start();
    byte[] event = encoder.encode(loggingEvent);

    System.out.println(new String(event));
```

**Comment by KarstenSchnitter** [comment]
Hi,

thank you for reporting the issue. I will look into it.

Best Regards,
Karsten

**Comment by KarstenSchnitter** [comment]
I am wondering though, why this is not picked up by the unit test case https://github.com/SAP/cf-java-logging-support/blob/456492122afd09625649753e13699b4ad804be16/cf-java-logging-support-logback/src/test/java/com/sap/hcp/cf/logging/common/TestAppLog.java#L125-L139
I will look into your example and the test.

**Comment by KarstenSchnitter** [comment]
I will run a comparison between 3.5.x and 3.6.1 to see what got lost.

**Comment by KarstenSchnitter** [comment]
I provided a fix with #146. It brings back the implementation from before v3.6.0, which is also used in the log4j2. The test was changed to detect future regressions.

When logging exceptions, keep in mind, that the information contained in the exceptions themselves depends on the JVM. The old test would not contain a message in the NullPointerException raised when running on an OpenJDK-11, but it would contain a message on a SAPMachine-17.
